lib/matplotlib/sankey.py

```diff
@@ -443,12 +442,14 @@ def add(self, patchlabel='', flows=None, orientations=None, labels='',
             # In the code below, angles are expressed in deg/90.
             rotation /= 90.0
         if orientations is None:
-            orientations = [0, 0]
-        if len(orientations) != n:
+            orientations = 0
+        try:
+            orientations = np.broadcast_to(orientations, n)
+        except ValueError:
             raise ValueError(
-            "orientations and flows must have the same length.\n"
-            "orientations has length %d, but flows has length %d."
-            % (len(orientations), n))
+                f"The shapes of 'flows' ({np.shape(flows)}) and "
+                f"'orientations' ({np.shape(orientations)}) are incompatible"
```

Comment:
`np.shape` returns a tuple, so the parentheses are probably not needed as they'd be doubled.